src/components/PendingTransactionsSnackbar/index.js

```diff
@@ -160,9 +90,79 @@ export default class PendingTransactionsSnackbar extends Component {
         return undefined;
       }
     }
-  };
+  }
 
-  onCloseClicked = () => {
-    this.props.togglePendingTxsSnackbar(false);
-  };
+  componentDidMount() {
+    this.props.store.pendingTxsSnackbar.init();
+  }
+
+  render() {
+    const {
+      isVisible,
+      count,
+      pendingCreateEvents,
+      pendingBets,
+      pendingSetResults,
+      pendingVotes,
+      pendingFinalizeResults,
+      pendingWithdraws,
+      pendingTransfers,
+      pendingResetApproves,
+    } = this.props.store.pendingTxsSnackbar;
+    const { classes, intl } = this.props;
+
+    const pendingCounts = {
+      [TransactionType.CreateEvent]: pendingCreateEvents,
+      [TransactionType.Bet]: pendingBets,
+      [TransactionType.SetResult]: pendingSetResults,
+      [TransactionType.Vote]: pendingVotes,
+      [TransactionType.FinalizeResult]: pendingFinalizeResults,
+      [TransactionType.Withdraw]: pendingWithdraws,
+      [TransactionType.Transfer]: pendingTransfers,
+      [TransactionType.ResetApprove]: pendingResetApproves,
+    };
+
+    if (count === 0) {
+      return null;
+    }
+
+    return (
+      <Snackbar
+        className={classes.snackbar}
+        open={isVisible}
+        anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+        message={
+          <Grid container>
+            <Grid item xs={11}>
+              <Typography variant="caption" className={classes.totalCountText}>
+                <FormattedMessage
+                  id="pendingTxsSnackbar.youHaveXPendingTransactions"
+                  defaultMessage="You have {numOfTxs} pending transaction(s)."
+                  values={{ numOfTxs: count }}
+                />
+              </Typography>
+              {
```

Comment:
Can we do it this way please.
```js
const pendingCounts = {
  'str.createEvent': pendingCreateEvents,
  'str.bet': pendingBets,
  'str.setResult': pendingSetResults,
  'str.vote': pendingVotes,
  'str.finalizeResult': pendingFinalizeResults,
  'str.withdraw': pendingWithdraws,
  'str.transferTokens': pendingTransfers,
  'tx.resetApproval': pendingResetApproves,
}

{Object.entries(pendingCounts).map(([ id, amounts ])) => amounts.length > 0 && (
  <Typography variant="caption" key={id}>{`${intl.formatMessage({ id })}: ${amounts.length}`}</Typography>
))}
```
Then we can remove `messages/defineMessages` and `this.getEventName`

Reply:
Also, please refer to the [style guide](https://bodhi.frontify.com/document/230177#/react-best-practices) on frontify. I added `harrison@bodhi.network` to it so you should be able to see it.

Reply:
done